contracts/modules/staking/StakingPool.sol

```diff
@@ -663,32 +662,42 @@ contract StakingPool is IStakingPool, ERC721 {
     );
 
     {
-      require(request.rewardRatio <= REWARDS_DENOMINATOR, "StakingPool: reward ratio exceeds denominator");
+      require(config.rewardRatio <= REWARDS_DENOMINATOR, "StakingPool: reward ratio exceeds denominator");
 
-      uint rewards = premium * request.rewardRatio / REWARDS_DENOMINATOR;
+      uint rewards = premium * config.rewardRatio / REWARDS_DENOMINATOR;
       uint expireAtBucket = Math.divCeil(block.timestamp + request.period, BUCKET_DURATION);
       uint _rewardPerSecond = rewards / (expireAtBucket * BUCKET_DURATION - block.timestamp);
 
       // 1 SLOAD + 1 SSTORE
       rewardBuckets[expireAtBucket].rewardPerSecondCut += _rewardPerSecond;
 
-      // scale back from 2 to 18 decimals
-      allocatedCoverAmount *= NXM_PER_ALLOCATION_UNIT;
-
-      // premium and rewards already have 18 decimals
-      return (allocatedCoverAmount, premium, rewards);
+      // apply the global rewards ratio and the total Rewards in NXM
+      // TODO: implement me
+      // tokenController().mintStakingPoolNXMRewards(rewardsInNXM, allocationRequests[i].poolId);
```

Comment:
what's the reason for not implementing this in this PR?

Reply:
I'm actually intending to move this call to the StakingPool itself. There're too many items going back and forth. It will be slightly more expensive but the amount of staking pools used in a cover will usually be small (probably most of the time just one) so calling directly from the pool might simplify things. It's entirely possible that the implementation may get unexpectedly complicated in which case I'll move it back here and implement.